packages/flame_behavior_tree/behavior_tree/lib/src/node.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:behavior_tree/behavior_tree.dart';
+
+/// The valid values for status of a node.
+enum NodeStatus {
+  /// Indicates that the node has not been ticked yet.
+  notStarted,
+
+  /// Indicates that the node is running.
+  running,
+
+  /// Indicates that the node has completed successfully.
+  success,
+
+  /// Indicates that the node has failed.
+  failure,
+}
+
+/// An interface which all the nodes implement.
+///
+/// Some examples are [Selector], [Sequence], [Inverter] and [Limiter].
+abstract interface class INode {
```

Comment:
`NodeInterface` maybe? Capital `I` for interfaces feels more Java than Dart

Reply:
Renamed to `NodeInterface` 👍